packages/nvidia_nat_core/src/nat/front_ends/console/console_front_end_plugin.py

```
@@ -38,9 +40,31 @@ async def prompt_for_input_cli(question: InteractionPrompt) -> HumanResponse:
+            ready, _, _ = select.select([sys.stdin], [], [], 1)
+            if ready:
+                user_response: str = sys.stdin.readline().strip()
+                return HumanResponseText(text=user_response)
+            remaining -= 1
+            # Save cursor position, update countdown line, restore cursor position
+            sys.stdout.write(f"\033[s\033[A\r[{remaining}s remaining]\033[K\033[u")
+            sys.stdout.flush()
+
+        error_msg: str = question.content.error or "This prompt is no longer available."
+        click.echo(f"\n{Fore.RED}{error_msg}{Fore.RESET}")
+        raise TimeoutError(f"HITL prompt timed out after {timeout}s waiting for human response")
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)